contrib/submit-protocol/src/App/ProtocolForm.tsx

```diff
@@ -0,0 +1,227 @@
+                          onChange={this.importFile}
+                        />
+                      </label>
+                      <DefaultButton text="View/Edit" onClick={this.openEditor} />
+                  </div>
+                </div>
+                <div className="modal-footer" style={{ marginTop: "256px" }}>
+                  <PrimaryButton text="Submit Job" onClick={this.submitProtocol} />
+                </div>
+              </div>
+            </form>
+          </div>
+        </Fabric>
+      </>
+    );
+  }
+
+  private setJobName = (jobName: string) => {
+    const protocol = this.state.protocol;
+    protocol.name = jobName;
+    this.setState({ jobName });
+    this.setState({ protocol });
+    this.setState({ protocolYAML: jsyaml.safeDump(protocol) });
+  }
+
+  private importFile = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const files = event.target.files;
+    if (!files || !files[0]) {
+      return;
+    }
+    const fileReader = new FileReader();
+    fileReader.addEventListener("load", () => {
+      const text = fileReader.result as string;
+      try {
+        const protocol = jsyaml.safeLoad(text);
+        this.setState({ jobName: protocol.name || "" });
+        this.setState({ protocol });
+        this.setState({ protocolYAML: text });
+      } catch (err) {
+        window.alert(err.message);
+      }
+    });
+    fileReader.readAsText(files[0]);
+  }
+
+  private editProtocol = (text: string) => {
+    this.setState({ protocolYAML: text });
+  }
+
+  private openEditor = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    this.setState({ showEditor: true });
+  }
+
+  private closeEditor = () => {
+    this.setState({ showEditor: false });
+  }
+
+  private saveEditor = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    const text = this.state.protocolYAML;
+    try {
+      const protocol = jsyaml.safeLoad(text);
+      this.setState({ jobName: protocol.name || "" });
+      this.setState({ protocol });
+      this.setState({ showEditor: false });
+    } catch (err) {
+      window.alert(err.message);
+    }
+  }
+
+  private discardEditor = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    const protocol = this.state.protocol;
+    this.setState({ protocolYAML: jsyaml.safeDump(protocol) });
+    this.setState({ showEditor: false });
+  }
+
+  private submitProtocol = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    if (this.state.protocolYAML == null) {
+      return;
+    }
+    window.fetch(`${this.props.api}/api/v2/jobs`, {
+      body: this.state.protocolYAML,
+      headers: {
+        "Authorization": `Bearer ${this.props.token}`,
+        "Content-Type": "text/yaml",
+      },
+      method: "POST",
+    }).then((res) => {
+      return res.json();
```

Comment:
Maybe you need to parse the result manually if yaml is returned.

Reply:
The result is a json response like:
```js
{
  "status": "202",
  "messgae": "..."
}
```
Only the config file from user is yaml.